### PR TITLE
feat: add `+` button back to create workspace

### PR DESCRIPTION
**Pull request by keita-determined**
## Description

WEB-840

Related: #5644

## Test Plan
- if user have a permission to create workspaces, user can see the `+` button in nav bar  
- click the `+` button in nav bar
- check if user can create a workspace

## Commentary (optional)
@alexjdetermined agree to have this back.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by keita-determined** [comment]
@mapmeld 
Talked to @michielpr [ref](https://determined-ai.slack.com/archives/C04FJLZHBEG/p1675281851919929?thread_ts=1675276549.376969&cid=C04FJLZHBEG), and updated the PR


https://user-images.githubusercontent.com/109113616/216203619-ab7a9d6c-c430-4d58-b9da-8bd7f8704f62.mov

**Review by mapmeld — APPROVED** [review]
Thanks for checking in with design

File: webui/react/src/components/NavigationSideBar.module.scss [hunk 1]
```diff
@@ -37,6 +37,9 @@
     overflow: hidden;
 
     .actionButtons {
+      display: none;
```
Comment (mapmeld) [inline webui/react/src/components/NavigationSideBar.module.scss:40]: these are always hidden?

Comment (mapmeld) [reply webui/react/src/components/NavigationSideBar.module.scss:40]: or I guess it would be helpful if you add a comment here

**Comment by keita-determined** [comment]
I'll merge this cuz react-test ci has been flaky and not related to this PR. 
Additionally, #5900 would fix the issue

**Pull request merged**